src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.Cache.cs

```diff
@@ -444,16 +444,10 @@ public static ulong GetKey(ReadOnlySpan<byte> propertyName)
             {
                 key = MemoryMarshal.Read<ulong>(propertyName);
 
-                // Max out the length byte.
-                // This will cause the comparison logic to always test for equality against the full contents
-                // when the first 7 bytes are the same.
-                key |= 0xFF00000000000000;
-
-                // It is also possible to include the length up to 0xFF in order to prevent false positives
-                // when the first 7 bytes match but a different length (up to 0xFF). However the extra logic
-                // slows key generation in the majority of cases:
-                // key &= 0x00FFFFFFFFFFFFFF;
-                // key |= (ulong) 7 << Math.Max(length, 0xFF);
+                key &= 0x00FFFFFFFFFFFFFF;
+
+                // Include the length with a max of 0xFF so we force comparisons when length >=0xFF.
+                key |= (ulong)7 << Math.Min(length, 0xFF);
```

Comment:
This helps with cache misses as it avoids the SequenceEquals if the lengths are different.

The extra time for key generation (to call Math.Min) was not significant enough in the fast-path to overweight the benefits of helping with cache misses.